File: src/main/java/org/stt/command/CommandHandler.java
package org.stt.command;

import org.stt.model.TimeTrackingItem;

import java.util.Collection;

public interface CommandHandler {
    void addNewActivity(NewActivity command);

    void endCurrentActivity(EndCurrentItem command);

    void removeActivity(RemoveActivity command);

    void removeActivityAndCloseGap(RemoveActivity command);

    void resumeActivity(ResumeActivity command);

    void resumeLastActivity(ResumeLastActivity command);

    void bulkChangeActivity(Collection<TimeTrackingItem> itemsToChange, String activity);
}

File: src/main/kotlin/org/stt/command/CommandHandler.kt
package org.stt.command

import org.stt.model.TimeTrackingItem

interface CommandHandler {
    fun addNewActivity(command: NewActivity)

    fun endCurrentActivity(command: EndCurrentItem)

    fun removeActivity(command: RemoveActivity)

    fun removeActivityAndCloseGap(command: RemoveActivity)

    fun resumeActivity(command: ResumeActivity)

    fun resumeLastActivity(command: ResumeLastActivity)

    fun bulkChangeActivity(itemsToChange: Collection<TimeTrackingItem>, activity: String)
}
